asteroid/engine/gan_system.py

```diff
@@ -0,0 +1,172 @@
+from collections import OrderedDict
+import soundfile as sf
+import pytorch_lightning as pl
+import torch
+import os
+from torch_stoi import NegSTOILoss
+from asteroid.losses import PITLossWrapper
+
+
+class GanSystem(pl.LightningModule):
+
+    """ Base class for GAN training systems.
+       Contains a generator a discriminator, two optimizers,
+       a generator loss function, a discriminator loss function,
+       a validation loss function,
+       training and validation dataloaders and learning rate scheduler.
+
+       Args:
+           discriminator (torch.nn.Module): Instance of discriminator (d).
+           generator (torch.nn.Module): Instance of generator (g).
+           opt_d (torch.optim.Optimizer): Instance of optimizer for d.
+           opt_g (torch.optim.Optimizer): Instance of optimizer for g.
+           discriminator_loss (callable): Loss function with signature
+               (est_targets, targets).
+            generator_loss (callable): Loss function with signature
+               (est_targets, targets).
+           train_loader (torch.utils.data.DataLoader): Training dataloader.
+           val_loader (torch.utils.data.DataLoader): Validation dataloader.
+           scheduler_d (torch.optim.lr_scheduler._LRScheduler): Instance
+               of learning rate schedulers for d.
+           scheduler_g (torch.optim.lr_scheduler._LRScheduler): Instance
+               of learning rate schedulers for g.
+           conf: Anything to be saved with the checkpoints during training.
+               The config dictionary to re-instantiate the run for example.
+       """
+
+    def __init__(self, discriminator, generator, opt_d, opt_g,
+                 discriminator_loss, generator_loss, train_loader,
+                 validation_loss=PITLossWrapper(NegSTOILoss(), pit_from='pw_pt'),
```

Comment:
I would put `None` as the validation loss, and we don't run validation step if we don't have validation loss. 
Imposing STOI doesn't seem right to me.